File: Cpp/IExecutable.cpp
#include "IExecutable.h"

IExecutable::IExecutable()
{
}


IExecutable::~IExecutable()
{
}

void IExecutable::execute(int num)
{
	switch (num)
	{
	case 1:
		question1();
		break;
	case 2:
		question2();
		break;
	case 3:
		question3();
		break;
	case 4:
		question4();
		break;
	default:
		break;
	}
}

void IExecutable::question1()
{
}

void IExecutable::question2()
{
}

void IExecutable::question3()
{
}

void IExecutable::question4()
{
}

File: Nim/IExecutable.nim
type IExecutable* = ref object of RootObj

method question1*(this: IExecutable) {.base.} =
    return

method question2*(this: IExecutable) {.base.} =
    return

method question3*(this: IExecutable) {.base.} =
    return

method question4*(this: IExecutable) {.base.} =
    return

method execute*(this: IExecutable, num: int) {.base.} =
    case num
    of 1:
        this.question1
    of 2:
        this.question2
    of 3:
        this.question3
    of 4:
        this.question4
    else:
        discard